src/cosmosdb-preview/azext_cosmosdb_preview/_params.py

```diff
@@ -153,6 +153,80 @@ def load_arguments(self, _):
     with self.argument_context('managed-cassandra datacenter list') as c:
         c.argument('cluster_name', options_list=['--cluster-name', '-c'], help="Cluster Name", required=True)
 
+    # Mongo Cluster
+    with self.argument_context('cosmosdb mongocluster') as c:
+        c.argument('cluster_name', completer=None, options_list=['--cluster-name', '-c'], help='Name of the Cosmos DB Mongo Cluster.', id_part=None)
+        c.argument('resource_group', completer=None, options_list=['--resource-group', '-g'], help='Name of the resource group of mongo cluster.', id_part=None)
+        c.argument('location', options_list=['--location', '-l'], help="Azure Location of the Cluster", required=True)
+
+    with self.argument_context('cosmosdb mongocluster create') as c:
+        c.argument('cluster_name', completer=None, options_list=['--cluster-name', '-c'], help='Name of the Cosmos DB Mongo Cluster.', id_part=None)
+        c.argument('resource_group', completer=None, options_list=['--resource-group', '-g'], help='Name of the resource group of mongo cluster.', id_part=None)
+        c.argument('location', options_list=['--location', '-l'], help="Azure Location of the Cluster", required=True)
+        c.argument('administrator_login', options_list=['--administrator-login'], help="The initial administrator user to be configured when a cluster is created", required=True)
+        c.argument('administrator_login_password', options_list=['--administrator-login-password'], help="The initial administrator password to be configured when a cluster is created", required=True)
+        c.argument('server_version', options_list=['--server-version'], help="The server version of the mongo cluster", required=True)
+        c.argument('shard_node_tier', options_list=['--shard-node-tier'], help="The node tier for mongo cluster.", required=True, arg_group='Shard')
+        c.argument('shard_node_ha', arg_type=get_three_state_flag(), options_list=['--shard-node-ha'], help="If enabled the cluster has HA.", required=True, arg_group='Shard')
+        c.argument('shard_node_disk_size_gb', options_list=['--shard-node-disk-size-gb'], help="The node disk size for mongo cluster.", required=True, arg_group='Shard')
+        c.argument('shard_node_count', options_list=['--shard-node-count'], help="The node count for mongo cluster.", required=True, arg_group='Shard')
+        c.argument('restore_point_in_time_utc', options_list=['--restore-point-in-time-utc'], action=UtcDatetimeAction, help="The timestamp to which the account has to be restored to.", arg_group='Restore')
+        c.argument('restore_source_resource_id', options_list=['--restore-source-resource-id'], help="The restorable source Id of the source mongo cluster from which the cluster has to be restored", arg_group='Restore')
```

Comment:
Two concerns here:
1. Restore parameters need to include createMode = 'PointInTimeRestore' on the create cluster request.
2. I think `restore` should be a separate command entirely like how Cosmos DB account restore works. This takes the restore parameters and then handles mappping the createMode = to 'PointInTimeRestore'.

Given this and that restore is not yet supported in prod, my suggestion is to not include the restore commands for the time bieng.

Reply:
fixed